src/tests/tests/background.js

```diff
@@ -357,4 +358,53 @@ QUnit.test("subdomains on the yellowlist are preserved", (assert) => {
   );
 });
 
+QUnit.test("migrates misattributed MDFP domains from snitch map", (assert) => {
+  setupBadgerStorage(badger);
+
+  // store one action map for all tests (should always be preserved)
+  const actionMap = badger.storage.getBadgerStorageObject('action_map'),
+    snitchMap = badger.storage.getBadgerStorageObject('snitch_map');
+
+  // reset snitchMap's storage to blank object
+  snitchMap.updateObject({});
+
+  // snitch map with no MDFP entries
+  let snitchNoMDFP = {
+    ['amazon.com']: ['amazonads.com', 'amazing.com', 'amazonrainforest.com']
+  };
+
+  // snitch map with some MDFP entries
+  let snitchSomeMDFP = {
+    ['amazon.com']: ['amazon.ca', 'amazon.co.jp', 'amazing.com']
+  };
+
+  // snitch map with all MDFP entries
+  let snitchAllMDFP = {
+    ['amazon.com']: ['amazon.ca', 'amazon.co.jp', 'amazon.es']
+  };
+
```

Comment:
We should either overwrite (and afterwards restore) actual MDFP data with our test data, or we should confirm that the assumptions we make about the contents of MDFP hold true. The latter seems a bit easier. Something like the following?
```javascript
  // confirm our MDFP assumptions
  ['amazon.ca', 'amazon.co.jp', 'amazon.es'].forEach(function (domain) {
    assert.ok(
      mdfp.isMultiDomainFirstParty('amazon.com', domain),
      domain + " is indeed MDFP to amazon.com"
    );
  });
```